torch/csrc/jit/interpreter.cpp

```diff
@@ -434,6 +444,13 @@ struct CodeImpl {
   void insertInstruction(OpCode op, int64_t X = 0, uint64_t N = 0) {
     instructions_.emplace_back(op, X, N);
     instructions_source_.emplace_back(current_node_);
+    if (torch::autograd::profiler::hasCallbacks()) {
+      std::string debug_info_str;
+      debug_info_str = current_node_->sourceRange().text();
+      replaceAllChar('\n', debug_info_str);
+      int64_t graph_id = graph_->getGraphId();
+      torch::autograd::profiler::setSourceDebugInfo(graph_id, debug_info_str);
```

Comment:
btw, the contract of appending the instructions in order is pretty fragile

Reply:
@dzhulgakov, you are right. I followed onto what is already assumed. I still do agree with you. If the implementation of how instructions are stored changes it requires corresponding change here too. Let me think how to address this better.